r-package/grf/tests/testthat/test_average_effect.R

```diff
@@ -363,3 +376,35 @@ test_that("average effect estimation doesn't error on data with a single feature
   average_partial_effect(forest)
   expect_true(TRUE) # so we don't get a warning about an empty test
 })
+
+test_that("average conditional local average treatment effect estimation is reasonable", {
+    p <- 10
+    n <- 1000
+    
+    X <- matrix(2 * runif(n * p) - 1, n, p)
+    A <- rnorm(n)
+    Z <- rbinom(n, 1, 0.5)
+    W <- A + Z * (1 + (X[,2] > 0))
+    tau <- X[,1] > 0
+    Y <- 2 * (X[,1] <= 0) * A + tau * W + (1 + (sqrt(3) - 1) * (X[,1] > 0)) * rnorm(n)
+    
+    forest.iv <- instrumental_forest(X, Y, W, Z, num.trees = 250)
+    compliance.forest <- causal_forest(forest.iv$X.orig,
+                                       Y=forest.iv$W.orig,
+                                       W=forest.iv$Z.orig,
+                                       Y.hat=forest.iv$W.hat,
+                                       W.hat=forest.iv$Z.hat,
+                                       sample.weights = forest.iv$sample.weights,
+                                       num.trees = 250)
+    compliance.score <- predict(compliance.forest)$predictions
+    
+    tau.hat <- average_late(forest.iv, compliance.score=compliance.score)
+    tau.x1p <- average_late(forest.iv, compliance.score=compliance.score,
+                            subset = X[,1] > 0)
+    
+    expect_equal(as.numeric(tau.hat["estimate"]), mean(tau), tol = 0.2)
```

Comment:
The probability of failure for at least one of these tests is 4.3% in 1000 simulations. Do you think that's too high? (I think it's right at the boundary of acceptable, will leave it up to you).

Reply:
It's probably OK -- but we can re-visit the thresholds if it causes trouble.